src/background/helpers.js

```diff
@@ -31,6 +33,22 @@ chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
     case 'openPrivateWindowWithUrl':
       chrome.windows.create({ url: msg.url, incognito: true });
       break;
+    case 'idle': {
+      // The idle detection is a two-step process:
+      // Schedule delayed check to ensure that the background process has started
+      // running the tasks. Then resolve all options observers and send the response
+      // in the next delayed timeout.
+      setTimeout(() =>
+        resolveObservers().then(() => {
+          setTimeout(() => {
```

Comment:
setImmidiate?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate 
> Deprecated: This feature is no longer recommended. Though some browsers might still support it, it may have already been removed from the relevant web standards...

The `requestIdleCallback` is not available in Service Workers - I tried it...